source/h5vol/H5VolUtil.c

```diff
@@ -0,0 +1,232 @@
+#include "H5VolUtil.h"
+#include "H5VolError.h"
+
+void *safe_calloc(size_t n, size_t s, unsigned long line)
+{
+    if (n == 0)
+    {
+        return NULL;
+    }
+
+    void *p = calloc(n, s);
+    if (!p)
+    {
+        fprintf(stderr, "[%s:%ld]Out of memory at calloc (%ld, %ld)\n",
+                __FILE__, line, (unsigned long)n, (unsigned long)s);
+        exit(EXIT_FAILURE);
```

Comment:
Do you really want to call `exit` here and terminate the entire program?

Reply:
Well, yes, nothing will succeed when there is not enough memory